src/main/java/software/amazon/awssdk/sar/ClientBuilderUsingRetryStrategy.java

```diff
@@ -0,0 +1,154 @@
+package software.amazon.awssdk.sar;
+
+import software.amazon.awssdk.awscore.retry.AwsRetryStrategy;
+import software.amazon.awssdk.core.client.config.ClientOverrideConfiguration;
+import software.amazon.awssdk.core.exception.SdkException;
+import software.amazon.awssdk.core.internal.retry.RetryPolicyAdapter;
+import software.amazon.awssdk.core.internal.retry.SdkDefaultRetryStrategy;
+import software.amazon.awssdk.core.retry.RetryMode;
+import software.amazon.awssdk.core.retry.RetryPolicy;
+import software.amazon.awssdk.core.retry.RetryUtils;
+import software.amazon.awssdk.retries.AdaptiveRetryStrategy;
+import software.amazon.awssdk.retries.api.BackoffStrategy;
+import software.amazon.awssdk.retries.api.RetryStrategy;
+import software.amazon.awssdk.services.dynamodb.DynamoDbClient;
+
+/**
+ * Building clients using {@link RetryStrategy} is similar to using a {@link RetryPolicy}.
+ */
+public class ClientBuilderUsingRetryStrategy {
+    /**
+     * Using retry mode is handy and supported by
+     * {@link ClientOverrideConfiguration.Builder#retryStrategy(RetryMode)}
+     */
+    public static DynamoDbClient clientWithRetryStrategy01() {
+        return DynamoDbClient
+                .builder()
+                .overrideConfiguration(o -> o.retryStrategy(RetryMode.STANDARD))
+                .build();
+    }
+
+    /**
+     * The new {@link RetryMode#ADAPTIVE} is <em>also</em> supported by retry strategies, but we don't have a concrete
+     * implementation of it, instead behind the scenes it uses the policy to strategy internal adapter
+     * {@link RetryPolicyAdapter} built to support backwards compatibility but not expected to be used by customers (it
+     * is an internal API).
+     */
+    public static DynamoDbClient clientWithRetryStrategy03() {
+        // Does *not* throws UnsupportedOperationException.
+        return DynamoDbClient
+                .builder()
+                .overrideConfiguration(o -> o.retryStrategy(RetryMode.ADAPTIVE))
+                .build();
+    }
+
+    /**
+     * RetryStrategy is an interface parametrized on its concrete implementation and its builder, the full signature of
+     * the type is
+     * {@snippet
+     * public interface RetryStrategy<B extends CopyableBuilder<B, T> & RetryStrategy.Builder<B, T>,
+     *                                T extends ToCopyableBuilder<B, T> & RetryStrategy<B, T>>
+     *                      extends ToCopyableBuilder<B, T> { ⋯
```

Comment:
What would it change to not have RetryStrategy be parameterized? This is fairly complicated, and doesn't feel super useful to track this information.

Reply:
I know, we cannot we should not declare `T build()` on the builder, and `B toBuilder()` on the strategy.

Reply:
Will research into this.